packages/lib/src/data-grid/utils.tsx

```diff
@@ -0,0 +1,387 @@
+import DxcCheckbox from "../checkbox/Checkbox";
+import DxcFlex from "../flex/Flex";
+import { HalstackProvider } from "../HalstackContext";
+import DxcIcon from "../icon/Icon";
+import { GridColumn, HierarchyGridRow, GridRow, ExpandableGridRow } from "./types";
+import { Column, RenderSortStatusProps, SortColumn, textEditor } from "react-data-grid";
+
+const checkboxTheme = {
+  checkbox: {
+    backgroundColorChecked: "#ffffff",
+    hoverBackgroundColorChecked: "#e6e6e6",
+    disabledBackgroundColorChecked: "#999999",
+    readOnlyBackgroundColorChecked: "#999999",
+    hoverReadOnlyBackgroundColorChecked: "#808080",
+    borderColor: "#ffffff",
+    hoverBorderColor: "#ffffff",
+    disabledBorderColor: "#999999",
+    readOnlyBorderColor: "#999999",
+    hoverReadOnlyBorderColor: "#808080",
+    checkColor: "#5f249f",
+    disabledCheckColor: "#ffffff",
+    readOnlyCheckColor: "#ffffff",
+    fontFamily: "Open Sans, sans-serif",
+    fontSize: "0.875rem",
+    fontWeight: "400",
+    fontColor: "#ffffff",
+    disabledFontColor: "#999999",
+    focusColor: "#0095ff",
+    checkLabelSpacing: "8px",
+  },
+};
+
+// Column<any, any> type added to avoid conflicts with SelectColumn typing from RDG
+export const convertToRDGColumns = (gridColumn: GridColumn, summaryRow?: GridRow): Column<any, any> => {
+  return {
+    key: gridColumn.key,
+    name: gridColumn.name,
+    resizable: gridColumn.resizable,
+    sortable: gridColumn.sortable,
+    draggable: gridColumn.draggable,
+    editable: gridColumn.textEditable,
+    headerCellClass: gridColumn.alignment ? `align-${gridColumn.alignment}` : `align-left`,
+    renderEditCell: gridColumn.textEditable ? textEditor : undefined,
+    renderCell: ({ row }) => {
+      return (
+        <div className={`ellipsis-cell ${gridColumn.alignment ? "align-" + gridColumn.alignment : "align-left"}`}>
+          {row[gridColumn.key]}
+        </div>
+      );
+    },
+    renderSummaryCell: () => {
+      return gridColumn.summaryKeyToRender ? (
+        <div className={`ellipsis-cell ${gridColumn.alignment ? "align-" + gridColumn.alignment : "align-left"}`}>
+          {summaryRow?.[gridColumn.summaryKeyToRender]}
+        </div>
+      ) : undefined;
+    },
+  };
+};
+
+export const renderSortStatus = ({ sortDirection }: RenderSortStatusProps) => {
+  return (
+    <div className="sortIconContainer">
+      {sortDirection !== undefined ? (
+        sortDirection === "ASC" ? (
+          <DxcIcon icon="Keyboard_Arrow_Up" />
+        ) : (
+          <DxcIcon icon="Keyboard_Arrow_Down" />
+        )
+      ) : (
+        <DxcIcon icon="Expand_All" />
+      )}
+    </div>
+  );
+};
+
+export const renderHierarchyTrigger = (
+  rows: HierarchyGridRow[],
+  triggerRow: HierarchyGridRow,
+  uniqueRowId: string,
+  columnKey: string,
+  // eslint-disable-next-line no-unused-vars
+  onRowsToRender: (rows: HierarchyGridRow[]) => void
+) => {
+  return (
+    <button
```

Comment:
aria-expanded it is not a valid parameter in button elements.
There are some problems about what value we assignt to aria-label, because the value inside the cell can be another component, so we can not use that.